src/HCSR04.cpp

```diff
@@ -20,21 +22,30 @@ double UltraSonicDistanceSensor::measureDistanceCm() {
 }
 
 double UltraSonicDistanceSensor::measureDistanceCm(float temperature) {
+    uint32_t maxDistanceDurationMicroSec;
+
     // Make sure that trigger pin is LOW.
     digitalWrite(triggerPin, LOW);
     delayMicroseconds(2);
     // Hold trigger for 10 microseconds, which is signal for sensor to measure distance.
     digitalWrite(triggerPin, HIGH);
     delayMicroseconds(10);
     digitalWrite(triggerPin, LOW);
+    double speedOfSoundInCmPerMicroSec = 0.03313 + 0.0000606 * temperature; // Cair ≈ (331.3 + 0.606 ⋅ ϑ) m/s
+
+    // Compute max delay based on max distance with 25% margin in microseconds or use absolute timeout
+    maxDistanceDurationMicroSec = 2.5 * maxDistanceCm / speedOfSoundInCmPerMicroSec;
+    if (timeoutMicroSec > 0) {
+    	min(maxDistanceDurationMicroSec, timeoutMicroSec);
+    }
+
     // Measure the length of echo signal, which is equal to the time needed for sound to go there and back.
-    unsigned long durationMicroSec = pulseIn(echoPin, HIGH);
+    unsigned long durationMicroSec = pulseIn(echoPin, HIGH, maxDistanceDurationMicroSec); // can't measure beyond max distance
```

Comment:
This is returning unsigned long -> should we maybe also use unsigned long for microsec then, instead of uint32_t?

Reply:
unsigned long is equivalent to uint32_t but this type is clearer for devs so I changed it to uint32_t to keep every variables to be of the same type.